services/services.go

```diff
@@ -76,21 +76,21 @@ func (s *serviceSharding) SetNumShards(n int) ServiceSharding { s.numShards = n;
 func NewServiceInstance() ServiceInstance { return new(serviceInstance) }
 
 type serviceInstance struct {
+	service  ServiceID
 	id       string
-	service  string
-	zone     string
 	endpoint string
 	shards   shard.Shards
 }
 
-func (i *serviceInstance) Service() string                          { return i.service }
-func (i *serviceInstance) ID() string                               { return i.id }
-func (i *serviceInstance) Zone() string                             { return i.zone }
-func (i *serviceInstance) Endpoint() string                         { return i.endpoint }
-func (i *serviceInstance) Shards() shard.Shards                     { return i.shards }
-func (i *serviceInstance) SetService(s string) ServiceInstance      { i.service = s; return i }
-func (i *serviceInstance) SetID(id string) ServiceInstance          { i.id = id; return i }
-func (i *serviceInstance) SetZone(z string) ServiceInstance         { i.zone = z; return i }
+func (i *serviceInstance) ServiceID() ServiceID { return i.service }
+func (i *serviceInstance) InstanceID() string   { return i.id }
+func (i *serviceInstance) Endpoint() string     { return i.endpoint }
+func (i *serviceInstance) Shards() shard.Shards { return i.shards }
+func (i *serviceInstance) SetServiceID(service ServiceID) ServiceInstance {
```

Comment:
Can we put a blank line between this method and single-line methods above and below?